crates/amaru-stores/src/rocksdb/columns/dreps.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::rocksdb::common::{as_key, as_value, PREFIX_LEN};
+use amaru_ledger::store::StoreError;
+use rocksdb::Transaction;
+use tracing::error;
+
+use amaru_ledger::store::columns::dreps::{Key, Row, Value, EVENT_TARGET};
+
+/// Name prefixed used for storing DReps entries. UTF-8 encoding for "drep"
+pub const PREFIX: [u8; PREFIX_LEN] = [0x64, 0x72, 0x65, 0x70];
+
+/// Register a new DRep.
+pub fn add<DB>(
+    db: &Transaction<'_, DB>,
+    rows: impl Iterator<Item = (Key, Value)>,
+) -> Result<(), StoreError> {
+    for (credential, (anchor, deposit)) in rows {
+        let key = as_key(&PREFIX, &credential);
+
+        // In case where a registration already exists, then we must only update the underlying
```

Comment:
@KtorZ Looks to me that there would be some values in having a separated `update` methods in stores.